### PR TITLE
Validations and model specs

### PR DESCRIPTION
**Pull request by SaveryIV**
In this PR i added the tests for the models, and added the corresponding validations

### CODE REVIEW

File: spec/models/user_spec.rb
Comment (Lucas-Masaba) [inline spec/models/user_spec.rb:1]: - [ ] You have 11 failing tests. Please make sure all your written tests are passing.
image

**Review by Lucas-Masaba — CHANGES_REQUESTED** [review]
# Hello  @SaveryIV 👋 , 

Good job so far! 👍 
There are some issues that you still need to work on to go to the next project but you are almost there!

almost-there

## Highlights 🍾 
- Linter checks passing :heavy_check_mark: 
- Validations are working as expected :gear: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions 💭 
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @Lucas-Masaba in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by SaveryIV** [comment]
@Lucas-Masaba i can't check the tests anymore i recive a error named Migrations Pennding but i don't have any Migrations Pennding

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -0,0 +1,48 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  context '#create' do
+    it 'is valid with the existing name' do
+      expect(User.create(name: 'Lily')).to be_valid
+    end
+
+    it 'is not valid with a blank name' do
+      expect(User.create(name: nil)).to_not be_valid
```
Comment (Shedrack-Sunday) [inline spec/models/user_spec.rb:10]: - [ ] I figured out where the error is, which is making you see the need for migrations. In your `bin` directory, kindly change the executable file name to `ruby` from `ruby.exe` if you want to run the project in a Unix operating system, else, you can run it on Windows.

### Directory with the executable files:
image

- [ ] After successfully running your code, I saw a lot of test errors that you need to fix. Kindly first make sure you can run `rspec`, then fix those errors :

### Failed error tests:
image

**Review by Shedrack-Sunday — CHANGES_REQUESTED** [review]
Hello @SaveryIV     👋 

# CHANGES REQUIRED ♻️

Impressive work on the project so far👏
You've done extremely well so far in this project,  However, there are some adjustments you need to make to move to the next project. 💯



## Highlights   
- Good use of GitHub features ✔️   
- Nice work with Rails model✔️
- Kudus making the requirement asked      ✔️  

# - Required Changes  🌵  

- [ ] Kindly take note of the comments i made regarding your failing tests

# Optional suggestions 

Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing really improves your application.



Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: spec/models/user_spec.rb (hunk 1 above)
Comment (SaveryIV) [reply spec/models/user_spec.rb:10]: @Shedrack-Sunday Thank you a lot! I'm gonna try to fix it and i update you with new information

Comment (SaveryIV) [reply spec/models/user_spec.rb:10]: @Shedrack-Sunday I can't find the ruby file
image

Comment (SaveryIV) [reply spec/models/user_spec.rb:10]: image

Comment (SaveryIV) [reply spec/models/user_spec.rb:10]: @Shedrack-Sunday pls help me

File: spec/models/user_spec.rb [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    it 'returns three most recent posts' do
+      expect(@user.three_most_recent_posts.length).to eq 3
+    end
+
+    it 'returns the most recent posts with titles 3, 4, 5' do
+      titles = []
+      @user.three_most_recent_posts.each do |post|
+        titles.push(post.title.to_i)
+      end
+      expect(titles).to all(be_between(3, 5))
```
Comment (V-Blaze) [inline spec/models/user_spec.rb:45]: - You have done great so far :+1: :+1: But there are still some requirements you need to meet before moving to the next milestone. :+1: :100: 
- [ ] Looking at the screenshot below, you can see that you still have `11 test cases failing` I'll suggest you fix these. :+1:  

- - Screenshot reference:
 
Screenshot from 2023-11-03 10-06-21

## From your communication with the previous reviewer, you mentioned getting an error of `Pending Migration` even after running `rails db:migrate` As I have explained to you over the Zoom call, running `rails db:migrate` only migrates the development database and the test files makes use of the test database, so you need to run `rails db:migrate RAILS_ENV=test` to migrate the test database. I know I have helped you correct this but do take note of it in your subsequent projects. :smile: :small_airplane:

**Review by V-Blaze — CHANGES_REQUESTED** [review]
## Hi @SaveryIV 

# Almost there 🟡 

1094a3e620ebea4cd6cf2d0aceaa8f52


Great work so far, You have come a long way, keep moving forward👏 👍  
You've done well implementing some changes from the previous reviewers, but there are still some issues I think you need to improve on. 😄 👋 


 ## Highlights
-  Most tests are passing✔
- Nice communication with reviewers💯 🥇 ✔
- Correct git flow :heavy_check_mark: 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/models/user_spec.rb (hunk 1 above)
Comment (Shedrack-Sunday) [reply spec/models/user_spec.rb:10]: Hello @SaveryIV , please call me on zoom or slack if you're still confused

**Review by OyePriscilla — APPROVED** [review]
Hi @SaveryIV 👏 👏 👏 👏, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
All written tests passed ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**